service/src/builder.rs

```diff
@@ -80,209 +75,122 @@ fn extract_shuttle_toml_name(path: PathBuf) -> anyhow::Result<String> {
 pub async fn build_workspace(
     project_path: &Path,
     release_mode: bool,
-    tx: Sender<Message>,
+    _tx: Sender<Message>,
 ) -> anyhow::Result<Vec<BuiltService>> {
-    let (read, write) = pipe::pipe();
     let project_path = project_path.to_owned();
 
-    // This needs to be on a separate thread, else deployer will block (reason currently unknown :D)
-    tokio::task::spawn_blocking(move || {
-        trace!("started thread to to capture build output stream");
-        for message in Message::parse_stream(read) {
-            trace!(?message, "parsed cargo message");
-            match message {
-                Ok(message) => {
-                    if let Err(error) = tx.send(message) {
-                        error!("failed to send cargo message on channel: {error}");
-                    }
-                }
-                Err(error) => {
-                    error!("failed to parse cargo message: {error}");
-                }
-            }
-        }
-    });
```

Comment:
We should have something similar to this when getting output from `cargo` because otherwise we'll not be able to provide compilation data when doing local run [1] and [2] the compilation logs are also important for the deployer, because they're persisted and retrieved by user when accessing deployment logs or when they subscribe to the deployment logs websocket.

[1] https://github.com/shuttle-hq/shuttle/blob/d0b2f32cdfeba02b7fb145c11d8df57d0ad9a3ba/cargo-shuttle/src/lib.rs#L716
[2]

Reply:
It looks like getting the same behavior as the deleted code block is a pain, but it looks like I could use [duct](https://crates.io/crates/duct/0.13.6) to do that.

Reply:
I think we don't need to have the exact same logic. I think we can try to redirect the `cargo` process stderr/stdout to the parent process stderr/stdout for local run while for deployer, we should make sure we extract both the stdout and stderr (separated or concatenated) into the deployer persistence.

Reply:
I've made a bit of progress. I'm using `os_pipe` instead of `duct`, to minimize the number of needed dependencies. I'm basically just doing [the second example](https://github.com/oconnor663/os_pipe.rs#examples).